components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMServiceTest.java

```diff
@@ -578,17 +581,23 @@ public void updateApplicationTestWithException()
         when(mockApplicationManagementService.getServiceProvider(dummyClientName, dummyTenantDomain)).thenReturn
                 (serviceProvider);
 
-        OAuthConsumerAppDTO appDTO = new OAuthConsumerAppDTO();
-        appDTO.setApplicationName(dummyClientName);
+        when(mockApplicationUpdateRequest.getClientName()).thenReturn(dummyClientName);
+        doNothing().when(mockApplicationManagementService).updateApplication(serviceProvider, dummyTenantDomain,
+                dummyUserName);
+        when(mockApplicationUpdateRequest.getGrantTypes()).thenReturn(dummyGrantTypes);
+        when(mockApplicationUpdateRequest.getRedirectUris()).thenReturn(redirectUri);
 
         doThrow(new IdentityOAuthAdminException("")).when(mockOAuthAdminService).updateConsumerApplication(any
                 (OAuthConsumerAppDTO.class));
 
         try {
+            startTenantFlow();
             dcrmService.updateApplication(mockApplicationUpdateRequest, dummyClientId);
         } catch (IdentityException ex) {
             assertEquals(ex.getErrorCode(), DCRMConstants.ErrorMessages.FAILED_TO_UPDATE_APPLICATION.toString());
             return;
```

Comment:
Rather than doing this this way, a better approach would be, catch the exception and then do the assert and re-throw the exception. Declare the exception type as an expected exception.